web/src/admin/roles/RelatedRoleList.ts

```diff
@@ -0,0 +1,185 @@
+                                    this.requestUpdate();
+                                }}
+                            >
+                                ${role.name}
+                            </ak-chip>`;
+                        })}
+                    </ak-chip-group>
+                </div>
+            </div>
+        </ak-form-element-horizontal>`;
+    }
+}
+
+@customElement("ak-role-related-list")
+export class RelatedRoleList extends Table<Role> {
+    checkbox = true;
+    clearOnRefresh = true;
+    protected override searchEnabled = true;
+
+    @property()
+    order = "name";
+
+    @property({ attribute: false })
+    targetUser?: User;
+
+    async apiEndpoint(): Promise<PaginatedResponse<Role>> {
+        return new RbacApi(DEFAULT_CONFIG).rbacRolesList({
+            ...(await this.defaultEndpointConfig()),
+            users: this.targetUser ? [this.targetUser.pk] : [],
+        });
+    }
+
+    protected columns: TableColumn[] = [
+        [msg("Name"), "name"],
+        [msg("Actions"), null, msg("Row Actions")],
+    ];
+
+    renderToolbarSelected(): TemplateResult {
+        const disabled = this.selectedElements.length < 1;
+        return html`<ak-forms-delete-bulk
+            objectLabel=${msg("Role(s)")}
+            actionLabel=${msg("Remove from Role(s)")}
+            actionSubtext=${msg(
+                str`Are you sure you want to remove user ${this.targetUser?.username} from the following roles?`,
+            )}
+            buttonLabel=${msg("Remove")}
+            .objects=${this.selectedElements}
+            .delete=${(item: Role) => {
+                if (!this.targetUser) return;
+                return new RbacApi(DEFAULT_CONFIG).rbacRolesRemoveUserCreate({
+                    uuid: item.pk,
+                    userAccountSerializerForRoleRequest: {
+                        pk: this.targetUser.pk,
+                    },
+                });
+            }}
+        >
+            <button ?disabled=${disabled} slot="trigger" class="pf-c-button pf-m-danger">
+                ${msg("Remove")}
+            </button>
+        </ak-forms-delete-bulk>`;
+    }
+
+    row(item: Role): SlottedTemplateResult[] {
+        return [
+            html`<a href="#/identity/roles/${item.pk}">${item.name}</a>`,
+            html` <ak-forms-modal>
+                <span slot="submit">${msg("Update")}</span>
+                <span slot="header">${msg("Update Role")}</span>
+                <ak-role-form slot="form" .instancePk=${item.pk}> </ak-role-form>
+                <button slot="trigger" class="pf-c-button pf-m-plain">
+                    <pf-tooltip position="top" content=${msg("Edit")}>
+                        <i class="fas fa-edit" aria-hidden="true"></i>
+                    </pf-tooltip>
+                </button>
+            </ak-forms-modal>`,
+        ];
+    }
+
+    renderToolbar(): TemplateResult {
+        return html`
+            ${this.targetUser
+                ? html`<ak-forms-modal>
```

Comment:
Short circuit pattern:
```suggestion
        if (!this.targetUser) {
          return nothing
```